packages/react/src/index.js

```diff
@@ -72,108 +59,96 @@ function Microlink (props) {
       setLoading(true)
       const fetch = isFunction(setData)
         ? Promise.resolve({})
-        : fetchFromApi(props.url, apiUrl, apiUrlProps)
-
-      fetch.then(({ data }) => mergeData(data))
-    }
-  }, [apiUrl, canFetchData, setData, apiUrlProps.headers['x-api-key']])
-
-  const mergeData = useCallback(fetchData => {
-    const payload = isFunction(setData)
-      ? setData(fetchData)
-      : { ...fetchData, ...setData }
-
-    const { title, description, url, video, audio, image, logo } = payload
-
-    const mediaFallback = image || logo || {}
-    let media = mediaFallback
-    let videoUrl
-    let audioUrl
-    let isVideo = false
-    let isAudio = false
-
-    if (!isNil(audio)) {
-      isAudio = true
-      audioUrl = getUrlPath(audio)
-    } else if (!isNil(video)) {
-      isVideo = true
-      videoUrl = getUrlPath(video)
-    } else {
-      media = someProp(payload, [].concat(props.media)) || mediaFallback
+        : fetchFromApi(url, apiUrl, apiUrlProps)
+
+      fetch
+        .then(({ data }) => mergeData(data))
+        .catch(error => {
+          setLoading(false)
+          setIsError(true)
+          console.warn('[Microlink SDK]', error)
+        })
```

Comment:
This `catch` and the `isError` flag resolves #154 too. I could imagine an end-user preferring no card at all instead of some "error UI".